src/test/java/com/powsybl/openloadflow/ac/AreaInterchangeControlTest.java

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright (c) 2024, Coreso SA (https://www.coreso.eu/) and TSCNET Services GmbH (https://www.tscnet.eu/)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.ac;
+
+import com.powsybl.iidm.network.Area;
+import com.powsybl.iidm.network.Network;
+import com.powsybl.loadflow.LoadFlow;
+import com.powsybl.loadflow.LoadFlowParameters;
+import com.powsybl.loadflow.LoadFlowResult;
+import com.powsybl.math.matrix.DenseMatrixFactory;
+import com.powsybl.openloadflow.OpenLoadFlowParameters;
+import com.powsybl.openloadflow.OpenLoadFlowProvider;
+import com.powsybl.openloadflow.network.*;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.concurrent.CompletionException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Valentin Mouradian {@literal <valentin.mouradian at artelys.com>}
+ */
+class AreaInterchangeControlTest {
+
+    private LoadFlow.Runner loadFlowRunner;
+    private LoadFlowParameters parameters;
+
+    private OpenLoadFlowParameters parametersExt;
+
+    @BeforeEach
+    void setUp() {
+        loadFlowRunner = new LoadFlow.Runner(new OpenLoadFlowProvider(new DenseMatrixFactory()));
+        parameters = new LoadFlowParameters();
+        parametersExt = OpenLoadFlowParameters.create(parameters)
+                .setAreaInterchangeControl(true)
+                .setSlackBusPMaxMismatch(1e-3);
+    }
+
+    @Test
+    void twoAreasWithXnodeTest() {
+        Network network = MultiAreaNetworkFactory.createTwoAreasWithXNode();
+        runLfTwoAreas(network, -40, 40, -30, 2);
+    }
```

Comment:
Usability comment:
   If you create an unbalanced situation (for example -40, 41 as  interchange targets), it is difficult to troubleshoot.
  The solver fails with max iteration reached. I have seen no other warning.

   I wonder if it would not help users to add some checks, at least when all boundaries belong each to at least two areas.

Reply:
Closing this one since a new parameter exists now for area mismatch